test/vagrant/vagrant_test.go

```diff
@@ -16,6 +17,11 @@ import (
 )
 
 func TestVagrantSetupGuide(t *testing.T) {
+	_, ok := os.LookupEnv("TEST_WITH_VAGRANT")
```

Comment:
can also add a build constraint right? I believe that works for test files too. Not sure if thats any better than this env look up though tbh.